### PR TITLE
add ETL option to parallelize transformers and loaders

### PR DESCRIPTION
**Pull request by calvinnhieu**
Adds 2 inputs to `extract` to control callback parallelism:
- `max_dispatchers` level of parallelism
- `dispatch_executor_class` specifies a `concurrent.futures.Executor` to use

### CODE REVIEW

File: dcplib/etl/__init__.py [hunk 1]
```diff
@@ -133,15 +134,17 @@ def extract(self, query: dict = None, transformer: callable = None, loader: call
...
-        for future in concurrent.futures.as_completed(ff_futures):
-            f, bundle_uuid, bundle_version = future.result()
-            self.b2f[bundle_uuid + "." + bundle_version].remove(f["name"])
-            if len(self.b2f[bundle_uuid + "." + bundle_version]) == 0:
-                self.dispatch_callbacks(bundle_uuid, bundle_version, transformer, loader)
-            else:
-                logger.debug("%s: %d files to go", bundle_uuid, len(self.b2f[bundle_uuid + "." + bundle_version]))
-        for bundle_uuid, bundle_version in self.staged_bundles:
-            self.dispatch_callbacks(bundle_uuid, bundle_version, transformer, loader)
+        with dispatch_executor_class(max_workers=max_dispatchers) as executor:
```
Comment (calvinnhieu) [inline dcplib/etl/__init__.py]: Is there a performance concern running this with `max_workers=1`?

Comment (kislyuk) [reply dcplib/etl/__init__.py]: Not sure if I follow. Here, max_workers=1 is the same as no multithreading at all, right?

**Comment by kislyuk** [comment]
Looks like there's a small linter error:
```
dcplib/etl/__init__.py:113:69: E252 missing whitespace around parameter equals
dcplib/etl/__init__.py:113:70: E252 missing whitespace around parameter equals
```

File: dcplib/etl/__init__.py (hunk 1 above)
Comment (calvinnhieu) [reply dcplib/etl/__init__.py]: Yea, unnecessary but leaving as is for code simplicity

**Pull request merged**